docs/try-flink/table_api.md

```diff
@@ -462,14 +469,17 @@ public class SpendReport {
         tEnv.registerTableSource("transactions", new UnboundedTransactionTableSource());
         tEnv.registerTableSink("spend_report", new SpendReportTableSink());
 
-        tEnv
+        Table table = tEnv
             .scan("transactions")
             .window(Tumble.over("1.hour").on("timestamp").as("w"))
             .groupBy("accountId, w")
-            .select("accountId, w.start as timestamp, amount.sum")
-            .insertInto("spend_report");
+            .select("accountId, w.start as timestamp, amount.sum");
 
-        env.execute("Spend Report");
+        // trigger execution
+        TableResult tableResult = table.executeInsert("spend_report");
+        // wait job finished
+        tableResult.getJobClient().get()
```

Comment:
side comment: This API looks very ugly and we should not document it like this. I also saw in our tests and didn't like it. How about we introduce a `TableResult.awaitCompletion(timeout)`? For DCL and DDL, it would continue immediately.

Reply:
Big +1 for this. We also have the same feeling and had some discussion around this. Maybe we can also have a `TableResult.awaitCompletion()` for this.

Reply:
Thanks for the feedback. agree with you that current api is not friendly for testing or demos.  
For INSERT job, `TableResult.awaitCompletion(timeout)` will work fine. but for `SELECT` job, there are some problems: in current implementation, client will send a request to JM (which will re-forward the request to TM) to fetch a part of result when `Iterator.next()` is called. The sink operator has a fixed-length pool to buffer the result. If the pool is full, the back-pressure mechanism will trigger. So the job will not finish until all data has been fetched to client (or the job is canceled).  Code snippet:

```
TableResult tableResult = ...
TableResult.awaitCompletion(timeout); 
CloseableIterator<Row> it = TableResult.collect();
it...
```

`TableResult.awaitCompletion(timeout)` will block the execution of `TableResult.collect()`. This will cause the job will not finish, or get TimeoutException thrown by `awaitCompletion` method.

Reply:
Yes, I think the behavior is fine. Maybe we can express it in a better name and definition. How about `TableResult.await()` that "waits until a result is ready"? Either the result is ready through `collect` or for INSERT INTO the result is ready once the job is completed and the number of rows is returned (at least this is the case for batch and would be the same for streaming in the future).

Reply:
make sense to me